pyvo/dal/tap.py

```diff
@@ -664,12 +673,15 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         """
-        Exits the context. The job is silently deleted.
+        Exits the context. Unless delete=False was set at initialization,
+        the job is deleted. Any deletion errors are silently ignored
+        to ensure proper context exit.
         """
-        try:
-            self.delete()
-        except Exception:
-            pass
+        if self._delete_on_exit:
+            try:
+                self.delete()
+            except Exception:
```

Comment:
This may point beyond the PR, but while we're at it, maybe use a more specific exception here rather than a blanket one?